webapp/channels/src/components/channel_header_mobile/unmute_channel_button/unmute_channel_button.tsx

```diff
@@ -9,36 +9,28 @@ import {NotificationLevels} from 'utils/constants';
 
 type Actions = {
     updateChannelNotifyProps: (userId: string, channelId: string, props: ChannelNotifyProps) => void;
-}
+};
 
 type Props = {
     user: { id: string };
     channel: { id: string };
     actions: Actions;
-}
-
-export default class UnmuteChannelButton extends React.PureComponent<Props> {
-    handleClick = (): void => {
-        const {
-            user,
-            channel,
-            actions: {
-                updateChannelNotifyProps,
-            },
-        } = this.props;
-
-        updateChannelNotifyProps(user.id, channel.id, {mark_unread: NotificationLevels.ALL} as ChannelNotifyProps);
+};
+
+const UnmuteChannelButton = ({user, channel, actions}: Props) => {
+    const handleClick = () => {
+        actions.updateChannelNotifyProps(user.id, channel.id, {mark_unread: NotificationLevels.ALL} as ChannelNotifyProps);
     };
 
-    render(): JSX.Element {
-        return (
-            <button
-                type='button'
-                className='navbar-toggle icon icon__mute'
-                onClick={this.handleClick}
-            >
-                <span className='fa fa-bell-slash-o icon'/>
-            </button>
-        );
-    }
-}
+    return (
+        <button
+            type='button'
+            className='navbar-toggle icon icon__mute'
+            onClick={handleClick}
+        >
+            <span className='fa fa-bell-slash-o icon'/>
+        </button>
+    );
+};
+
+export default UnmuteChannelButton;
```

Comment:
It used to be a pure component, so use here a `React.memo`.

Reply:
I have introduced memo for the component.

Reply:
Done!